src/parser.rs

```diff
@@ -108,43 +108,35 @@ define_parser!(ParseUnaryOp, UnaryOpKind, |_, state| {
 });
 
 lazy_static! {
-    static ref BINARY_OPS: HashMap<&'static str, BinaryOpKind> = {
+    static ref BINARY_OPS: HashMap<Symbol, BinaryOpKind> = {
         let mut map = HashMap::new();
 
-        map.insert(Symbol::Plus.to_str(), BinaryOpKind::Add);
-        map.insert(Symbol::Minus.to_str(), BinaryOpKind::Subtract);
-        map.insert(Symbol::Star.to_str(), BinaryOpKind::Multiply);
-        map.insert(Symbol::Slash.to_str(), BinaryOpKind::Divide);
-        map.insert(Symbol::Caret.to_str(), BinaryOpKind::Exponent);
-        map.insert(Symbol::TwoDots.to_str(), BinaryOpKind::Concat);
+        map.insert(Symbol::Plus, BinaryOpKind::Add);
+        map.insert(Symbol::Minus, BinaryOpKind::Subtract);
+        map.insert(Symbol::Star, BinaryOpKind::Multiply);
+        map.insert(Symbol::Slash, BinaryOpKind::Divide);
+        map.insert(Symbol::Caret, BinaryOpKind::Exponent);
+        map.insert(Symbol::TwoDots, BinaryOpKind::Concat);
 
         map
     };
```

Comment:
Can we just use a match statement inline instead of a big hashmap in this case?

Reply:
We could, yes! That'd be a lot simpler.